apps/system/js/nfc_handover_manager.js

```diff
@@ -14,689 +13,703 @@
-      this._restoreBluetoothStatus();
-      return;
-    }
+      window.addEventListener('bluetooth-adapter-added', function() {
+        self._debug('bluetooth-adapter-added');
+        var req = self.bluetooth.getDefaultAdapter();
+        req.onsuccess = function bt_getAdapterSuccess() {
+          self.settingsNotified = false;
+          self.defaultAdapter = req.result;
+          self._debug('MAC address: ' + self.defaultAdapter.address);
+          self._debug('MAC name: ' + self.defaultAdapter.name);
+          /*
+           * Call all actions that have queued up while Bluetooth
+           * was turned on.
+           */
+          for (var i = 0; i < self.actionQueue.length; i++) {
+            var action = self.actionQueue[i];
+            action.callback.apply(self, action.args);
+          }
+          self.actionQueue = [];
+        };
+        req.onerror = function bt_getAdapterError() {
+          self._logVisibly('event listner: Failed to get bluetooth adater');
+        };
+      });
 
-    var cps = this.bluetooth.enabled ? NDEF.CPS_ACTIVE : NDEF.CPS_ACTIVATING;
-    var mac = this.defaultAdapter.address;
-    var hs = NDEFUtils.encodeHandoverSelect(mac, cps);
-    var promise = nfcPeer.sendNDEF(hs);
-    promise.then(() => {
-      this._debug('sendNDEF(hs) succeeded');
-      this.incomingFileTransferInProgress = true;
-    }).catch(e => {
-      this._logVisibly('sendNDEF(hs) failed : ' + e);
-      this._clearTimeout();
-      this._restoreBluetoothStatus();
-    });
-
-    this._clearTimeout();
-    this.responseTimeoutFunction =
-      setTimeout(this._cancelIncomingFileTransfer.bind(this),
-                 this.responseTimeoutMillis);
-  },
-
-  /**
-   * Initiate a file transfer by sending a Handover Request to the
-   * remote device.
-   * @param {MozNFCPeer} An instance of MozNFCPeer object.
-   * @param {Blob} blob File to be sent.
-   * @param {String} requestId Request ID.
-   * @memberof NfcHandoverManager.prototype
-   */
-  _initiateFileTransfer:
-    function _initiateFileTransfer(nfcPeer, blob, requestId) {
-      this._debug('initiateFileTransfer');
-      /*
-       * Initiate a file transfer by sending a Handover Request to the
-       * remote device.
-       */
+      window.addEventListener('bluetooth-disabled', function() {
+        self._debug('bluetooth-disabled');
+        self._clearBluetoothStatus();
+      });
+
+      window.navigator.mozSetMessageHandler('nfc-manager-send-file',
+        function(msg) {
+          self._debug('In New event nfc-manager-send-file' +
+            JSON.stringify(msg));
+          self.handleFileTransfer(msg.peer, msg.blob, msg.requestId);
+      });
+
+      SettingsListener.observe('nfc.debugging.enabled', false,
+                               (enabled) => { this.DEBUG = enabled; });
+
+      window.addEventListener('nfc-transfer-started',
+        this._transferStarted.bind(this)
+      );
+      window.addEventListener('nfc-transfer-completed',
```

Comment:
Or keep it and call this.transferComplete in this._handle_nfc-transfer-completed